### ISSUE TITLE
Red Badge only works on "current" month

### ISSUE THREAD

**Issue by justbudget** [justbudget/justbudget#49]
The red badge can be confusing. It currently shows up if the _current_ month has any categories with a negative balance. This can be strange when, for example: 

If it's currently May, and no categories have a negative balance, I won't see any red dot. If I go into June and move money so that my "available to budget" is negative, I won't see a red dot. Then, as soon as it turns June 1st, I will see a red dot.

This can be confusing, because the red dot was intended to appear when entering transactions into an account, so that you would know that you need to re-work the budget.

**Comment by bpaulien** [comment]
I can see two sides to this.  One work flow that I've heard of is to keep the "next month" budget as a template, so technically the next month will be in the red (over budgeted) at the beginning of the month, but by the time all your income has come in this month, the next month is covered.  Then at the "month rollover" process, the next month budget is copied over to the next-next-month, and any adjustments needed for the now current month, are made, and balance of Available to budget would be brought to zero.

So in this case, that would always be red, because I had intentionally overbudgeted next month, since that's where my "template" is.  

Not saying you shouldn't count future months, it's probably a good idea.  Somebody using this workflow would just have to realize the red badge will always be there, and doesn't necessarily mean anything is 'wrong' with the budget.  I think I had at some point mentioned the possibility of having a separate "template"  that could be used instead of next month's budget.  And that's when the "copy previous budget" came in.

So here's a suggestion.  Make it user configurable for whether it looks into the future.  In the case of somebody using next month as the template, they could switch it off, because they already know they're over budgeted in the future.  It could default to on (looking into the future) because I don't think that's a "typical" workflow, and everybody else would be warned when some action today will affect the budget in the future.

**Comment by justbudget** [comment]
Thanks for the notes @bpaulien, I'm a little confused by your example, right now the dot never looks into the "future", it only looks into the _current_ month (as in, based on the clock, not based on what month is being viewed). So I'm not sure how it could be user configurable whether or not it looks into the future. 

I'm tempted to make the dot configurable entirely, since it's only supposed to be a convenience, not a nuisance.

**Comment by bpaulien** [comment]
I guess I mis-read what you were saying.  I like having the red dot, and I guess I'm not completely sure I understand the actual intended purpose of it, but here's how I at least would expect it to work, (maybe this is actually what it does, I haven't experimented with it enough to know what all activities trigger it) when any category in the current month goes red, it should show up red.  I'm not sure what the green is...  Is that when the Available to budget is positive, non-zero?  That's fine for me.  I mostly care about being notified if a category balance in the current month is negative.

Edited to add:  Really all I care about is current month.  If a negative balance passed over from last month, and I "fixed" it by transferring money to it in this current month, it should no longer indicate red, because that category "right now" is either 0 or positive.  I know that I should try to fix any red categories before the end of the month, but that doesn't always happen.  As long as it's dealt with in the current month, that's what I care about.

**Comment by justbudget** [comment]
> when any category in the current month goes red, it should show up red.

Exactly! That's how it works now. And just to reiterate, "current month" is based on the calendar, as in, if it's currently June, it will only look at June.

> Is that when the Available to budget is positive, non-zero? 

Yes this is exactly right. The intention here is if you enter in a pay check, you should go to the budget tab to assign your newly added "available to budget" to a category. I agree that the red dot is more useful.

>  Really all I care about is current month.

Yep I agree with that! That's how it's implemented now. If you have an overspent category last month, the red dot will not show up.
